client/src/components/FileDrop.jsx

```diff
@@ -25,9 +23,9 @@ const FileDrop = ({ setResponse = () => {} }) => {
     setMidiFile(null);
   };
 
-  const handleUpload = files => {
+  const handleUpload = (files) => {
     setDropFiles(files);
-    setMidiFile(files[0].file);
+    setMidiFile(files[0] ? files[0].file : null);
```

Comment:
there was an error happening here when you try to access the file property of files[0] when deleting since that's undefined.